server.go

```diff
@@ -254,6 +256,38 @@ func (rh *moduleHandler) handlePanic(wr http.ResponseWriter, req *http.Request,
 			return
 		}
 	} else {
+		writeErrorToResponse(err, debug.Stack(), wr)
 		log.Printf("[UncaughtPanic] %s\n-- ERROR --\nUncaught panic in route module %T: %+v\n-- STACK TRACE --\n%s", req.URL, rh.module, err, debug.Stack())
 	}
 }
+
+func writeErrorToResponse(errorString error, stackTrace []byte, wr http.ResponseWriter) {
+	tmpl := `
+    <html>
+        <head>
+            <title>Error Occured</title>
+        </head>
+        <body>
+            <style>
+                <h1>Error</h1>
+                <p>{{.Error}}</p>
+                <pre>{{.StackTrace}}</pre>
+            </style>
+        </body>
+    </html>
+    `
+
+	t := template.Must(template.New("error").Parse(tmpl))
```

Comment:
If this were to panic there is nothing higher up in the callstack to recover. Let's remove `template.Must` and catch the error. In the error block just move the log statement from above:
```go
log.Printf("[UncaughtPanic] %s\n-- ERROR --\nUncaught panic in route module %T: %+v\n-- STACK TRACE --\n%s", req.URL, rh.module, err, debug.Stack())
```